orchestra/models/core/models.py

```diff
@@ -523,3 +526,64 @@ class PayRate(PayRateMixin, models.Model):
     hourly_multiplier = models.DecimalField(max_digits=6, decimal_places=4)
     start_date = models.DateField(default=timezone.now)
     end_date = models.DateField(null=True, blank=True)
+
+
+class StaffingRequest(StaffingRequestMixin, models.Model):
+    """
+    A StaffingRequest object defines how a Worker was contacted to
+    work on a Task.
+
+    Attributes:
+        worker (orchestra.models.core.Worker):
+            Django user that the request is sent to
+        task (orchestra.models.core.Task):
+            A task that needs a worker assignment
+        request_cause (RequestCause):
+            A cause for request
+        project_description (str):
+            Description of a project
+    """
+
+    class RequestCause(ChoicesEnum):
+        USER = 'user'
+        AUTOSTAFF = 'autostaff'
+        RESTAFF = 'restaff'
+
+    worker = models.ForeignKey(Worker)
+    task = models.ForeignKey(Task)
+    request_cause = models.IntegerField(
+        choices=RequestCause.choices())
+    project_description = models.CharField(max_length=200,
+                                           null=True,
+                                           blank=True)
+
+
+class StaffingResponse(StaffingResponseMixin, models.Model):
+    """
+    A StaffingResponse object stores a response from a worker and
```

Comment:
If we don't store the communication type on the request, should we store it on the response?  the answer to this question should be based on whether we'll have one request per communication type, or one per request regardless of the number of communication types.